### PR TITLE
Allowed passing empty string for @dataAttributePrefix

### PR DESCRIPTION
**Pull request by ijlee2**
## Description

I forgot to test an edge case for `@dataAttributePrefix`. When a user passes an empty string (meaning they don't want a prefix for brevity), the data attributes should be in the form of `data-tall`, instead of `data--tall`.

### CODE REVIEW

File: addon/components/container-query.js [hunk 1]
```diff
@@ -49,7 +49,15 @@ export default class ContainerQueryComponent extends Component {
     const prefix = this.dataAttributePrefix;
 
     for (const [featureName, meetsFeature] of Object.entries(this.queryResults)) {
-      const attributeName = `data-${prefix}-${featureName}`;
+      let attributeName;
```
Comment (ijlee2) [inline addon/components/container-query.js:52]: An alternative approach to defining `attributeName` is:

```javascript
const attributeName = ['data', prefix, featureName]
  .filter(string => string)
  .join('-');
```

This is equivalent to the code that I committed in this PR, all except for when `featureName === ''` (an empty string is a valid key for POJOs).

Although this edge case (edge-edge case?) is unlikely to occur in practice, I decided to go with the `if-else` approach so that, if `featureName === ''`, then we will create a 1-1 mapping of data attributes. (The `filter-join` approach collapses two cases into 1.) The user will see the data attribute `data-`.
